packages/protocol/test-sol/integration/EpochManagerIntegration.sol

```diff
@@ -0,0 +1,247 @@
+      targetValidatorEpochPayment,
+      communityRewardFraction,
+      address(0),
+      carbonOffsettingFraction
+    );
+
+    epochManager.initialize(
+      REGISTRY_ADDRESS,
+      DAY,
+      carbonOffsettingPartner,
+      epochManagerInitializer
+    );
+
+    blockTravel(vm, firstEpochBlock);
+
+    // epochManager.
+  }
+}
+
+contract EpochManagerIntegrationTest_initialize is EpochManagerIntegrationTest {
+  function test_initialize() public virtual {
+    assertEq(address(epochManager.registry()), REGISTRY_ADDRESS);
+    assertEq(epochManager.epochDuration(), DAY);
+    assertEq(epochManager.carbonOffsettingPartner(), carbonOffsettingPartner);
+  }
+
+  function test_Reverts_WhenAlreadyInitialized() public virtual {
+    vm.expectRevert("contract already initialized");
+    epochManager.initialize(
+      REGISTRY_ADDRESS,
+      DAY,
+      carbonOffsettingPartner,
+      epochManagerInitializer
+    );
+  }
+}
+
+contract EpochManagerIntegrationTest_initializeSystem is EpochManagerIntegrationTest {
+  function test_processCanBeStarted() public virtual {
+    vm.prank(epochManagerInitializer);
+    epochManager.initializeSystem(firstEpochNumber, firstEpochBlock, firstElected);
+  }
+
+  function test_Reverts_processCannotBeStartedAgain() public virtual {
+    vm.prank(epochManagerInitializer);
+    epochManager.initializeSystem(firstEpochNumber, firstEpochBlock, firstElected);
+    vm.prank(address(0));
+    vm.expectRevert("Epoch system already initialized");
+    epochManager.initializeSystem(firstEpochNumber, firstEpochBlock, firstElected);
+  }
+
+  function test_Reverts_WhenSystemInitializedByOtherContract() public virtual {
+    vm.expectRevert("msg.sender is not Initializer");
+    epochManager.initializeSystem(firstEpochNumber, firstEpochBlock, firstElected);
+  }
+}
+
+contract EpochManagerIntegrationTest_startNextEpochProcess is EpochManagerIntegrationTest {
+  function test_Reverts_whenSystemNotInitialized() public {
+    vm.expectRevert("Epoch system not initialized");
+    epochManager.startNextEpochProcess();
+  }
+
+  function test_Reverts_WhenEndOfEpochHasNotBeenReached() public {
+    vm.prank(epochManagerInitializer);
+    epochManager.initializeSystem(firstEpochNumber, firstEpochBlock, firstElected);
+
+    vm.expectRevert("Epoch is not ready to start");
+    epochManager.startNextEpochProcess();
+  }
+
+  function test_Succeeds() public {
+    vm.prank(epochManagerInitializer);
+    epochManager.initializeSystem(firstEpochNumber, firstEpochBlock, firstElected);
+
+    blockTravel(vm, 43200);
+    timeTravel(vm, DAY);
+    console2.log("current epoch timestamp", block.timestamp);
+
+    uint256 _currentEpoch = epochManager.currentEpochNumber();
+    console2.log("### Current epoch duration:", epochManager.epochDuration());
+    (, , , uint256 _currentEpochEndTimestamp, uint256 _currentRewardsBlock) = epochManager
+      .getCurrentEpoch();
+
+    epochManager.startNextEpochProcess();
```

Comment:
Seems to be failing when calling `getCeloUnreleasedTreasure().release()` on the actual contract.
 not sure why